core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -655,45 +648,29 @@ func (r *EvmRegistry) doCheck(ctx context.Context, mercuryEnabled bool, keys []o
 	}
 
 	chResult <- checkResult{
-		ur: upkeepResults,
+		cr: upkeepResults,
 	}
 }
 
-func splitKey(key ocr2keepers.UpkeepKey) (*big.Int, *big.Int, error) {
-	var (
-		block *big.Int
-		id    *big.Int
-		ok    bool
-	)
-
-	parts := strings.Split(string(key), separator)
-	if len(parts) != 2 {
-		return nil, nil, fmt.Errorf("unsplittable key")
-	}
-
-	if block, ok = new(big.Int).SetString(parts[0], 10); !ok {
-		return nil, nil, fmt.Errorf("could not get block from key")
-	}
-
-	if id, ok = new(big.Int).SetString(parts[1], 10); !ok {
-		return nil, nil, fmt.Errorf("could not get id from key")
-	}
-
-	return block, id, nil
+func (r *EvmRegistry) getBlockAndUpkeepId(key ocr2keepers.UpkeepPayload) (*big.Int, *big.Int) {
+	block := new(big.Int).SetInt64(key.Trigger.BlockNumber)
```

Comment:
my understanding is for both conditional and log trigger upkeeps, this trigger block number will be populated.
for conditional, this will be the head and for log trigger, this will be the block where the log is emitted.

but should we use `CheckBlock` in the payload in some ways? i thought in recovery pipeline run, we will update the block key so it will be run on a new block?